src/linker/Linker.Dataflow/DynamicallyAccessedMembersTypeHierarchy.cs

```diff
@@ -210,14 +201,14 @@ public DynamicallyAccessedMembersTypeHierarchy (LinkContext context, MarkStep ma
 			if (applied) {
 				var origin = new MessageOrigin (type);
 				// Report warnings on access to annotated members, with the annotated type as the origin.
-				ApplyDynamicallyAccessedMembersToType (ref reflectionMarker, origin, type, annotation);
+				ApplyDynamicallyAccessedMembersToType (reflectionMarker, origin, type, annotation);
 				_typesInDynamicallyAccessedMembersHierarchy[type] = (annotation, true);
 			}
 
 			return applied;
 		}
 
-		void ApplyDynamicallyAccessedMembersToType (ref ReflectionMarker reflectionMarker, in MessageOrigin origin, TypeDefinition type, DynamicallyAccessedMemberTypes annotation)
+		void ApplyDynamicallyAccessedMembersToType (in ReflectionMarker reflectionMarker, in MessageOrigin origin, TypeDefinition type, DynamicallyAccessedMemberTypes annotation)
```

Comment:
Is it correct that we should use the RequireDynamicallyAccessedMembersAction.MarkTypeForDynamicallyAccessedMembers instead of the ReflectionMarker once we share this code with the analyzer?
(Maybe you thought about this... if not, no matter, somebody will look into it once we implement sharing of this).

Reply:
I actually didn't even think about that. I was thinking what it would take to share this code with the analyzer, which doesn't have a `ReflectionMarker`.